src/lib.rs

```diff
@@ -73,6 +73,29 @@ impl<T: Clone> Drop for Pinboard<T> {
     }
 }
 
+/// An wrapper around a `Pinboard` which provides the guarantee it is never empty.
+pub struct NonEmptyPinboard<T: Clone>(Pinboard<T>);
+
+impl<T: Clone> NonEmptyPinboard<T> {
+    /// Create a new `NonEmptyPinboard` instance holding the given value.
+    pub fn new(t: T) -> NonEmptyPinboard<T> {
+        NonEmptyPinboard(Pinboard::new(t))
+    }
+
+    /// Update the value stored in the `NonEmptyPinboard`.
+    pub fn set(&self, t: T) {
```

Comment:
I don't know  the rules for `inline` - [this answer](https://stackoverflow.com/a/37639889/5437033) seems to indicate that it isn't worth it for small functions?